runtime/lua/vim/_extui/shared.lua

```diff
@@ -77,6 +77,8 @@ function M.tab_check_wins()
     if setopt then
       local name = { cmd = 'Cmd', dialog = 'Dialog', msg = 'Msg', pager = 'Pager' }
       api.nvim_buf_set_name(M.bufs[type], ('[%s]'):format(name[type]))
+      api.nvim_set_option_value('swapfile', false, { buf = M.bufs[type] })
+      api.nvim_set_option_value('modifiable', true, { buf = M.bufs[type] })
```

Comment:
curious why this is necessary given we use `nvim_create_buf(false, true)` ? https://github.com/neovim/neovim/blob/927dc3c2c49faeb8be94097d5faa0322928059b1/runtime/lua/vim/_extui/shared.lua#L37

Reply:
`scratch = true` sets neither 'swapfile' nor 'modifiable', so they inherit the global value?

Reply:
It should, else it's a bug (`:help scratch-buffer`). This line is intended to set noswapfile: https://github.com/neovim/neovim/blob/5cf135f9a0ac472621d133bb4a03780a7bff6640/src/nvim/api/vim.c#L946

Reply:
Oh right. But yeah it also serves to restore the option values for those instances.

Reply:
Off topic, but I think we need a way to set a buffer as "permanently" scratch-like. This is a very common issue that makes scratch buffers less useful than they could be.

Reply:
Maybe 'swapfile', 'modifiable' (...?) should just be |local-noglobal|?